client/ayon_airtable/plugins/publish/collect_airtable_api.py

```diff
@@ -0,0 +1,64 @@
+
+"""Collects Airtable API and stores them in the context data."""
+
+import os
+
+import pyblish.api
+from ayon_airtable.common.BaseHandler import (
+    get_airtable_base,
+    get_airtable_table,
+    get_api_token,
+)
+
+
+class CollectAirtableAPI(pyblish.api.ContextPlugin):
+    """Collects Airtable API key and base name from the addon settings."""
+    order = pyblish.api.CollectorOrder + 0.1
+    label = "Collect Airtable API"
+    families = ["editorial", "editorial_pkg"]  # noqa: RUF012
+
+    def process(self, context: pyblish.api.Context) -> None:
+        """Collect Airtable API related info from environment variables.
+
+        Parameters
+        ----------
+        context : pyblish.api.Context
+            The Pyblish context to store Airtable API, base,
+            and table information.
+
+        Raises:
+        ------
+        RuntimeError
+            If collecting Airtable API information fails.
+        """  # noqa: DOC501
```

Comment:
Not sure why using different docstring style - I would stick with napoleon docstrings. Then you wouldn't need `DOC501`